daffodil-cli/build.sbt

```diff
@@ -71,3 +78,165 @@ rpmRelease := {
+// to this scheme.
+wixProductId := "F803A6CD-3008-4979-86BB-C6094AE57CD2"
+wixProductUpgradeId := "4C966AFF-585E-4E17-8CC2-059FD70FEC77"
+
+// Light options. Bring in standard dialog boxes and localization.
+// The suppression of ICE61 is required as we *DO* permit
+// re-installation of the same version. Despite the presence of
+// specific XML to enable this, the WiX compiler and linker
+// complain about it unless you specifically suppress the warning.
+lightOptions := Seq("-sice:ICE61", "-ext", "WixUIExtension", "-cultures:en-us", "-loc", sourceDirectory.value + "\\windows\\Product_en-us.wxl")
+
+// Point to the RTF license file so it will appear in the license
+// acceptance dialog box. BTW, this file was hand-sweetened to permit
+// flexible line/word wrap in support of variable display width as the
+// file will be rendered in both the (relatively narrow) dialog box
+// and ()optionally) sent to the (relatively wide) printer when the
+// user asks for hard copy.
+wixProductLicense := Option(file("daffodil-cli/src/windows/LICENSE.rtf"))
+
+// Use the wixFiles variable to add in the Daffodil-specific dialog
+// boxes and sequence.
+wixFiles ++= Seq(
+  file(sourceDirectory.value + "\\windows\\WixUI_DFDL.wxs"),
+  file(sourceDirectory.value + "\\windows\\DisclaimerDlg_DFDL.wxs")
+)  
+
+// The SBT WiX plug-in assumes that we want to give the user a
+// Feature Tree to select from. One of the 'features' that the plug-in
+// offers up is a set of all shortcuts and menu links. Daffodil is
+// actually a command-line executable, so we do not include
+// configuration links as they are unnecessary. From a practical
+// standpoint the user must invoke a command shell in a window
+// before they can invoke Daffodil anyway.
+wixFeatures := {
+  val features = wixFeatures.value
+  features.filter(f => f.id != "AddConfigLinks")
+}
+
+// Make sure that we don't use an MSI installer that is older than
+// version 4.0. It also fixes the comment attribute that hangs
+// out on the Package keyword. 
+wixPackageInfo := wixPackageInfo.value.copy(installerVersion = "400", comments = "!(loc.Comments)")
+
+// Fix the XML that is associated with the installable files and directories.
+wixProductConfig := {
+  // Pick up the generated code.
+  val pc = wixProductConfig.value
+  
+  // Replace the default headline banner and Welcome/Exit screen
+  // bitmaps with the custom ones we developed for Daffodil. 
+  val banner = <WixVariable Id="WixUIBannerBmp" Value={sourceDirectory.value + "\\windows\\daffodil_banner.bmp"} />
+  val splash = <WixVariable Id="WixUIDialogBmp" Value={sourceDirectory.value + "\\windows\\splash.bmp"} />
+  
+  // Reference the Daffodil-specific User Interface (dialog box) sequence.
+  val ui = <UI><UIRef Id="WixUI_DFDL" /></UI>
+  
+  // Make sure we abort if we are not installing on Windows Vista or later.
+  val oscondition = <Condition Message="!(loc.OS2Old)"><![CDATA[Installed OR (VersionNT >= 600)]]></Condition>
```

Comment:
Okay. Tell me how far back you want to support things and I'll change the limit to that.

Reply:
Honestly, I'm not really sure how far back is reasonable. Maybe limiting to Vista is? Or maybe no limits are reasonable? I guess I feel like we don't limit people from installing Daffodil to certain Linux systems, so what is to be gained by limiting installations to certain windows versions? If someone wants to try to install this on Windows 95, who am I to say no? We dont' depend on any windows capabilities specific to a certain windows versions, so any OS version should work as long as it supports the right java version.

Reply:
I *DO* still have my install disks for DOS 1.10 at home, should you need them.

Reply:
I'd like to think that we can test and debug on all the platforms we support. How else would you deal with the bug reports? Do we have access to installation disks for every release of Windows? How about DOS? At some point your argument about being OS-independent fails. For example, Windows 1.0 doesn't have a registry and DOS 1.10 doesn't even support tree-structured directories. Are we going to support command line operation on OS/2? On some older computers the hard disk isn't large enough to store the 29MB MSI file. I still have a 4.77MHz 8088-based computer with a 17MB external Corvus hard disk at home. How are you going to support me? At some point you just have to throw in the towel. What are your stated minimum system requirements?

Reply:
I have been thinking about this. I don't think the Windows MSI installer was even capable of supporting some of the operations we require prior to Windows 95. I'd bet we can't even install on Windows 3.0 (which runs on DOS 3.x). I think if we go back to limiting the MSI installer to version 2.0 ("200") then we can at least try to install on Windows 95 and above. What do you think?

Reply:
I think 200 seems like a reasonable limitation. And if a user tries to install it on something old like Windows 95 and it doesn't work, then we can always change things to be more restrictive. We certainly shouldn't limit it to only OS'es that we've tested on. Instead we should allow for the community to try to run it on whatever they want (within reason), and if we get a bug report that it doesn't work on someones OS of choice, we can then either fix it or say that vesion isn't supported. Limiting to Vista or newer just because we're not sure whether or not it works on XP/95 isn't a very convincing reason to me.

Reply:
I set OS limit to '200' and installer version limit to '200'.